File: frontend/src/types/job.ts
export type JobStatus = "new" | "applied" | "interviewing" | "rejected" | "offers" | "follow_up" | "skipped";

export type Job = {
  id: number;
  source: string;
  external_id: string | null;
  url: string;
  title: string;
  company: string;
  location: string | null;
  remote: boolean;
  description: string | null;
  requirements: string[];
  skills: string[];
  seniority: string | null;
  employment_type: string | null;
  score: number;
  score_breakdown: {
    skill_overlap?: number;
    experience_match?: number;
    location_match?: number;
    seniority_match?: number;
    tech_relevance?: number;
    matched_skills?: string[];
    missing_core_skills?: string[];
    notes?: string[];
  };
  status: JobStatus;
  rejection_reason: string | null;
  recruiter_name: string | null;
  recruiter_email: string | null;
  discovered_at: string;
  applied_at: string | null;
  updated_at: string;
};

export type Profile = {
  id: number;
  name: string;
  current_role: string | null;
  experience_years: number;
  skills: string[];
  preferred_roles: string[];
  preferred_locations: string[];
  resume_text?: string | null;
  profile_json?: Record<string, unknown>;
  source_resume_path: string | null;
};

export type ProfileUpdatePayload = {
  name?: string;
  current_role?: string | null;
  experience_years?: number;
  skills?: string[];
  preferred_roles?: string[];
  preferred_locations?: string[];
};

export type DashboardSummary = {
  new_jobs: number;
  applied: number;
  interviewing: number;
  rejected: number;
  offers: number;
  follow_ups: number;
  skipped: number;
  average_score: number;
  high_score_jobs: number;
};

export type Artifact = {
  id: number;
  job_id: number;
  kind: string;
  content: string;
  file_path: string | null;
  created_at: string;
};

export type ConnectorStatus = Record<string, { available: boolean; mode: string }>;

export type AutoApplyJobResult = {
  job_id: number;
  title: string;
  company: string;
  source: string;
  score: number;
  application_id: number | null;
  status: string;
  submitted: boolean;
  dry_run: boolean;
  message: string;
};

export type AutoApplyResponse = {
  processed: number;
  submitted: number;
  prepared: number;
  skipped: number;
  failed: number;
  dry_run: boolean;
  run_id: number | null;
  results: AutoApplyJobResult[];
};

File: backend/app/schemas/job.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class ScoreBreakdown(BaseModel):
    skill_overlap: int = 0
    experience_match: int = 0
    location_match: int = 0
    seniority_match: int = 0
    tech_relevance: int = 0
    matched_skills: list[str] = Field(default_factory=list)
    missing_core_skills: list[str] = Field(default_factory=list)
    notes: list[str] = Field(default_factory=list)


class JobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    source: str
    external_id: str | None
    url: str
    title: str
    company: str
    location: str | None
    remote: bool
    description: str | None
    requirements: list[str]
    skills: list[str]
    seniority: str | None
    employment_type: str | None
    score: int
    score_breakdown: dict[str, Any]
    status: str
    rejection_reason: str | None
    recruiter_name: str | None
    recruiter_email: str | None
    discovered_at: datetime
    applied_at: datetime | None
    updated_at: datetime


class JobListResponse(BaseModel):
    jobs: list[JobRead]
    total: int


class JobStatusUpdate(BaseModel):
    status: str
    notes: str | None = None


class DiscoveryRequest(BaseModel):
    sources: list[str] | None = None
    limit: int = 50


class DiscoveryResponse(BaseModel):
    discovered: int
    accepted: int
    skipped: int
    run_id: int | None = None

